pkg/provisioner/controller.go

```diff
@@ -462,33 +462,38 @@ func (c *obcController) supportedProvisioner(provisioner string) bool {
 	return provisioner == c.provisionerName
 }
 
-// Returns the ob, configmap, and secret based on the passed-in key. Only returns non-nil
-// error if unable to get all resources. Some resources may be nil.
-func (c *obcController) getResourcesFromKey(key string) (*v1alpha1.ObjectBucket, *corev1.ConfigMap, *corev1.Secret, error) {
-
-	ob, obErr := c.objectBucketForClaimKey(key)
-	if errors.IsNotFound(obErr) {
-		log.Error(obErr, "objectBucket not found")
-		obErr = nil
-	}
-	cm, cmErr := configMapForClaimKey(key, c.clientset)
-	if errors.IsNotFound(cmErr) {
-		log.Error(cmErr, "configMap not found")
-		cmErr = nil
-	}
-	s, sErr := secretForClaimKey(key, c.clientset)
-	if errors.IsNotFound(sErr) {
-		log.Error(sErr, "secret not found")
-		sErr = nil
+// trim the errors resulting from objects not being found
+func (c *obcController) getResourcesForDeletionByKey(key string) (*v1alpha1.ObjectBucket, *corev1.ConfigMap, *corev1.Secret, []error) {
+	ob, cm, secret, errs := c.getResourcesFromKey(key)
+	for i := len(errs) - 1; i >= 0; i-- {
+		if errors.IsNotFound(errs[i]) {
+			errs = append(errs[:i], errs[i+1:]...)
+		}
 	}
+	return ob, cm, secret, errs
+}
+
+// Gathers resources by names derived from key.
+// Returns pointers to those resources if they exist, nil otherwise and an slice of errors who's
+// len() == n errors. If no errors occur, len() is 0.
+func (c *obcController) getResourcesFromKey(key string) (ob *v1alpha1.ObjectBucket, cm *corev1.ConfigMap, sec *corev1.Secret, errs []error) {
 
 	var err error
-	// return err if all resources were not retrieved, else no error
-	if obErr != nil && cmErr != nil && sErr != nil {
-		err = fmt.Errorf("could not get all needed resources: %v : %v : %v", obErr, cmErr, sErr)
+	errs = make([]error, 0, 3)
```

Comment:
why 3 and why hard-coded?

Reply:
it's specific to the amount of possible errors that can occur locally.  3 separate API calls in this case.

Reply:
I see now. I suppose there's no point in defining a global _number of k8s resources_ const... Maybe just add a comment here?

Reply:
That's a good point, I'll drop one in here.

Reply:
fixed